packages/botonic-react/src/components/multichannel/multichannel-text.jsx

```diff
@@ -120,45 +132,104 @@ export const MultichannelText = props => {
       )
 
-      const messages = messagesPostbackButtons.map((postbackButtons, i) => {
-        if (i === 0) {
-          return [].concat(
-            ...textElements,
-            ...urlButtonElements,
-            ...postbackButtons
-          )
-        } else {
-          return [].concat(buttonsTextSeparator, ...postbackButtons)
+      const messages = messagesPostbackButtonList.map(
+        (postbackButtons, index) => {
+          if (postbackButtons.length <= WHATSAPP_MAX_BUTTONS) {
+            return {
+              type: INPUT.TEXT,
+              children: [...buttonsTextSeparator, ...postbackButtons],
+            }
+          }
+          const rows = postbackButtons.map(postbackButton => {
+            const row = {
+              id: postbackButton.props.path
+                ? `__PATH_PAYLOAD__${postbackButton.props.path}`
+                : postbackButton.props.payload,
+              title: postbackButton.props.children,
+            }
+            return row
+          })
+          const whatsbuttonlistProps = {
+            body: index === 0 ? textElements : buttonsTextSeparator,
+            button: menuButtonTextWhatsappList,
+            sections: [{ rows }],
+          }
+
+          return {
+            type: INPUT.WHATSAPP_BUTTON_LIST,
+            props: whatsbuttonlistProps,
+          }
         }
-      })
-      if (webviewButtonElements.length) {
-        messages.push([buttonsTextSeparator, ...webviewButtonElements])
-      }
+      )
+
+      const messageWithUrlButtonElements = (
+        <Text
+          key={`msg-with-url-button`}
+          {...MULTICHANNEL_WHATSAPP_PROPS}
+          {...props}
+        >
+          {urlButtonElements}
+        </Text>
+      )
+
+      const messageWithWebviewButtonElements = (
+        <Text
+          key={`msg-with-webview-button`}
+          {...MULTICHANNEL_WHATSAPP_PROPS}
+          {...props}
+        >
+          {buttonsTextSeparator}
+          {webviewButtonElements}
+        </Text>
+      )
 
       return (
         <>
-          {messages.map((message, i) => (
-            <Text key={i} {...MULTICHANNEL_WHATSAPP_PROPS} {...props}>
-              {message}
-            </Text>
-          ))}
+          {messages.map((message, i) => {
+            if (message.type === INPUT.WHATSAPP_BUTTON_LIST)
+              return (
+                <WhatsappButtonList
+                  key={`msg-${i}-whatsapp-list`}
+                  {...message.props}
+                />
+              )
+            return (
+              <Text
+                key={`msg-${i}-with-postback-buttons`}
+                {...MULTICHANNEL_WHATSAPP_PROPS}
+                {...props}
+              >
+                {message.children}
+              </Text>
+            )
+          })}
+          {urlButtonElements.length ? messageWithUrlButtonElements : null}
+          {webviewButtonElements.length
+            ? messageWithWebviewButtonElements
+            : null}
```

Comment:
Suggestion: Better use `&&`?
```suggestion
          {urlButtonElements.length && messageWithUrlButtonElements}
          {webviewButtonElements.length && messageWithWebviewButtonElements}
```

Reply:
Yes, but I should change the condition to length>0 because if it is an empty array it would render with 0 &&. message.....

Reply:
applied

Reply:
You're right!